### ISSUE TITLE
Section name in PEReader contains null padding

### ISSUE THREAD

**Issue by krwq** [dotnet/corefx#1805]
String in section name (i.e. peReader.PEHeaders.SectionHeaders[0].Name) contains null characters.

I believe we should remove it, unless it is by design.

ECMA description of field:
```
An 8-byte, null-padded ASCII string. There is no terminating null if the string is exactly eight characters long.
```

Here is how we get the value:
```csharp
// reader is PEBinaryReader
// _name is string
// PEFileConstants.SizeofSectionName = 8
_name = reader.ReadUTF8(PEFileConstants.SizeofSectionName);

// impl of ReadUTF8:
public string ReadUTF8(int byteCount)
{
    byte[] bytes = ReadBytes(byteCount);
    return Encoding.UTF8.GetString(bytes, 0, byteCount);
}
```

In theory it can also be encoding bug.

**Comment by ellismg** [comment]
Given the spec, it also seems a little odd that we are using a UTF8 decoder here.

**Comment by mikedn** [comment]
> Given the spec, it also seems a little odd that we are using a UTF8 decoder here.

Actually it's the ECMA spec that it is odd :smile:. The latest PECOFF spec says

> An 8-byte, null-padded UTF-8 encoded string...

A quick test shows that the VC++ compiler can indeed emit UTF-8 encoded section names.

**Comment by jbevain** [comment]
Wouldn't be the first time that the C++ compiler emitted something not conforming with ECMA-335 :)

**Comment by nguerrera** [comment]
Fixed by #1837
